file_management/include/file_management/file_upload/file_manager_strategy.h

```diff
@@ -301,6 +303,11 @@ class FileManagerStrategy : public DataManagerStrategy {
    */
   bool isDataAvailable() override;
 
+  /**
+   * Returns true if user set the option to remove older logs
+   */
+  bool discardOldLogs() override;
```

Comment:
```suggestion
  bool discardOldData() override;
```

If it was the log file manager the name would make sense, but misleading here otherwise. Same with the comment, maybe "discards stale data that won't be accepted by the cloudwatch API. Done in a separate method so it doesn't block the current publishing task after reading the data" etc. 

This way we clearly show intent. How does that sound?